spec/controllers/controller_patches_spec.rb

```diff
@@ -0,0 +1,193 @@
+# -*- encoding : utf-8 -*-
+# If defined, ALAVETELI_TEST_THEME will be loaded in config/initializers/theme_loader
+ALAVETELI_TEST_THEME = 'asktheeu-theme'
+require File.expand_path(File.join(File.dirname(__FILE__),
+                         '..','..','..','..', '..', 'spec','spec_helper'))
+
+describe GeneralController, "when patched by the asktheeu-theme" do
+
+  describe "caching blog content" do
+
+    before do
+      allow(controller).to receive(:blog).and_return("")
+    end
+
+    context "caching is disabled" do
+
+      before do
+        allow(AlaveteliConfiguration).to receive(:cache_fragments).
+          and_return(false)
+      end
+
+      it "does not attempt to read from the cache" do
+        expect(controller).not_to receive(:read_fragment)
+        controller.send(:blog_cache, "test", 1.minute)
+      end
+
+      it "calls the blog controller action" do
+        expect(controller).to receive(:blog)
+        controller.send(:blog_cache, "test", 1.minute)
+      end
+
+    end
+
+    context "the cache is empty" do
+
+      before do
+        allow(AlaveteliConfiguration).to receive(:cache_fragments).
+          and_return(true)
+        allow(controller).to receive(:read_fragment).with("test-updated").
+          and_return(nil)
+        allow(controller).to receive(:write_fragment)
+      end
+
+      context "the blog is available" do
+
+        before do
+          controller.instance_variable_set(:@blog_items, ["content"])
+        end
+
+        it "calls the blog controller action" do
+          expect(controller).to receive(:blog)
+          controller.send(:blog_cache, "test", 10.minutes)
+        end
+
+        it "sets the cache that holds the last updated timestamp" do
+          expect(controller).to receive(:write_fragment).
+            with("test-updated", anything)
+          controller.send(:blog_cache, "test", 10.minutes)
+        end
+
+      end
+
+      context "the blog is not available" do
+
+        before do
+          controller.instance_variable_set(:@blog_items, [])
+          allow(controller).to receive(:read_fragment).with("test-").
```

Comment:
`"test-updated"` is set in the `before do` block at the start of the containing section, which seemed like a good idea at the time but is really unclear